proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -325,34 +335,42 @@ public ProctorResult determineTestGroups(
         }
 
         final Map<String, ValueExpression> localContext =
-                ProctorUtils.convertToValueExpressionMap(
+                ProctorUtils.convertLocalContextToValueExpressionMap(
                         RuleEvaluator.EXPRESSION_FACTORY, inputContext);
 
         for (final String testName : filteredEvaluationOrder) {
             final TestChooser<?> testChooser = testChoosers.get(testName);
             final String identifier;
+            final boolean containsUnitlessAllocations =
+                    testTypesWithInvalidIdentifier.contains(
+                                    testChooser.getTestDefinition().getTestType())
+                            && testChooser.getTestDefinition().getContainsUnitlessAllocation();
+            localContext.put(
+                    UNITLESS_ALLOCATION_IDENTIFIER,
+                    containsUnitlessAllocations ? VALUE_EXPRESSION_TRUE : VALUE_EXPRESSION_FALSE);
             if (testChooser instanceof StandardTestChooser) {
                 final TestType testType = testChooser.getTestDefinition().getTestType();
-                if (testTypesWithInvalidIdentifier.contains(testType)) {
+                if (testTypesWithInvalidIdentifier.contains(testType)
+                        && !containsUnitlessAllocations) {
                     // skipping here to make it use the fallback bucket.
                     continue;
                 }
 
                 identifier = identifiers.getIdentifier(testType);
-                if (identifier == null) {
+                if (identifier == null && !containsUnitlessAllocations) {
                     // No identifier for the testType of this chooser, nothing to do
                     continue;
                 }
             } else {
-                if (!identifiers.isRandomEnabled()) {
+                if (!identifiers.isRandomEnabled() && !containsUnitlessAllocations) {
```

Comment:
if the test type is random, isRandomeEnabled is false, containsUnitlessAllocations is true, what will happen?

Reply:
i see it will go to standardchooser